src/studio-header/NavDropdownMenu.test.jsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import NavDropdownMenu from './NavDropdownMenu';
+
+const mockOnNavigate = jest.fn();
+
+const defaultProps = {
+  id: 'menu-id',
+  buttonTitle: 'Menu',
+  items: [
+    { href: '/item1', title: 'Item 1' },
+    { href: 'https://external.com', title: 'External Link' },
+  ],
+  onNavigate: mockOnNavigate,
+};
+
+describe('NavDropdownMenu Component', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test('renders the dropdown button with correct title', () => {
+    render(<NavDropdownMenu {...defaultProps} />);
+
+    const dropdownButton = screen.getByRole('button', { name: 'Menu' });
+    expect(dropdownButton).toBeInTheDocument();
+  });
+
+  test('renders all dropdown items', () => {
+    render(<NavDropdownMenu {...defaultProps} />);
+
+    const dropdownButton = screen.getByRole('button', { name: 'Menu' });
+    fireEvent.click(dropdownButton);
+
+    const item1 = screen.getByText('Item 1');
+    const externalLink = screen.getByText('External Link');
+
+    expect(item1).toBeInTheDocument();
+    expect(externalLink).toBeInTheDocument();
+  });
+
+  test('calls onNavigate with the correct URL for internal link', () => {
```

Comment:
All these "calls onNavigate with the correct URL for internal link" vs. "navigates to external URL when external link is clicked" tests are not really required if we get rid of `navigateToUrl` just use `react-router`'s `navigate` function, which has the same logic and is well-tested. Or if you use the `<Link>` component - you can just trust that those things work correctly.

Reply:
Implemented in follow-up commit: [`d5745e2` (#542)](https://github.com/openedx/frontend-component-header/pull/542/commits/d5745e210a9e7a57bb51ca229f4eebd00b365501)